### PR TITLE
CI: add Twine check in check workflow

### PR DESCRIPTION
**Pull request by armujahid**
Closes: #5655

### CODE REVIEW

**Comment by armujahid** [comment]
Working action can be checked at https://github.com/armujahid/scrapy/actions/runs/3168893838/jobs/5160421545

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -39,3 +39,8 @@ jobs:
       run: |
         pip install -U tox
         tox
+    - name: Twine check
+      run: |
+        pip install twine
+        python setup.py sdist
+        twine check dist/*
```
Comment (kmike) [inline .github/workflows/checks.yml:46]: Could you please move this check to tox.ini, similar to other checks?

Comment (kmike) [reply .github/workflows/checks.yml:46]: Currently this check would be run 5 times on CI, for flake8, typing, docs, etc. tox environments. We should make it a separate tox env, and run this env on CI, similar to how others are executed.

Comment (armujahid) [reply .github/workflows/checks.yml]: Done

**Review by kmike — APPROVED** [review]
👍 Thanks @armujahid!

**Comment by armujahid** [comment]
Windows/tests CI errors are unrelated

**Review by wRAR — APPROVED** [review]

**Pull request merged**